### ISSUE TITLE
Can’t establish a connection to the server at ws://wikimon.hatnote.com:9220/.

### ISSUE THREAD

**Issue by hugolpz** [hatnote/listen-to-wikipedia#48]
On Firefox : 

> Firefox can’t establish a connection to the server at ws://wikimon.hatnote.com:9220/.

![Screenshot from 2023-09-25 15-55-36](https://github.com/hatnote/listen-to-wikipedia/assets/1420189/976b8da9-2e4c-4cf5-9f10-06eb0d30225e)

On google chrome

> Failed to load resource: the server responded with a status of 404 (Not Found)

> WebSocket connection to 'ws://wikimon.hatnote.com:9000/' failed: WebSocket is closed before the connection is established.

![image](https://github.com/hatnote/listen-to-wikipedia/assets/1420189/2649e730-7443-475e-ac88-fcbcb47fd422)

**Comment by mahmoud** [comment]
Hey @hugolpz! So usually this issue is caused by local firewalls or network rules that block those ports and/or websockets, usually at a workplace/school setting. Our monitoring hasn't shown any connectivity outages from 5+ geographic locations in the last 24 hours. When you say the community has pointed out a widespread issue, can you share a link with more details? Thanks!

**Comment by hugolpz** [comment]
Wikimedia France just opened a [MOOC Wikidata](https://formations.wikimedia.fr/courses/course-v1:WikimediaFrance+MWd-1+2022/course/) . We mostly marketed it to universities. I'm indeed working at and using my university connection to access this page.  (Screenshot of forum to come.)

After testing with my personal cellphone as hotspot, I can confirm your assessment :
1. personal 5G connection : it works.
2. university connection : it stops. (see : no more Qid, only legacy dying out circle shapes from previous phase)

While the diagnosis is correct, maybe if now server answer for a while then a warning message to check our network firewall would be interesting. Should I open a ticket for this.


![Screenshot from 2023-09-26 09-26-55](https://github.com/hatnote/listen-to-wikipedia/assets/1420189/ee9402e6-9af0-470e-a7f9-27c0a3ee59b0)
![Screenshot from 2023-09-26 09-27-38](https://github.com/hatnote/listen-to-wikipedia/assets/1420189/d8993775-283b-4c8f-9ab7-30080abbacde)
![Screenshot from 2023-09-26 09-27-50](https://github.com/hatnote/listen-to-wikipedia/assets/1420189/0a8db523-4d55-49cf-8c45-21b0563e8936)

**Comment by mahmoud** [comment]
Yeah the best fix we identified for WS blockage is actually #31, which actually got a start in a rewrite by @xabriel this year: https://github.com/xabriel/listen-to-wikipedia/tree/vue-rewrite

A warning would be good in the meantime, if you'd like to submit a PR I'm happy to review. As far as the state of issues goes, I'd like to know more about what you're proposing, as so far L2W issue management hasn't been much trouble.
